engine/src/main/java/org/hibernate/validator/internal/engine/valueextraction/ValueExtractorManager.java

```diff
@@ -347,6 +351,124 @@ private static boolean isClassPresent(String className, boolean fallbackOnTCCL)
 		}
 	}
 
+	private final class ValueExtractorCache {
+
+		private final ConcurrentHashMap<ValueExtractorCacheKey, Set<ValueExtractorDescriptor>> valueExtractorCache;
```

Comment:
I don't think you should call it `valueExtractorCache`, it's rather disturbing as it's the name of the whole class.

Maybe `valueExtractorsByTypeArgument`? Not totally sold to it, maybe you have a better idea?